darshan-util/pydarshan/tests/test_plot_io_cost.py

```diff
@@ -12,140 +12,109 @@
     # regression test for `plot_io_cost.get_io_cost_df()`
-    actual_df = get_io_cost_df(report=report, mod_key=mod)
+    actual_df = get_io_cost_df(report=report)
     assert_frame_equal(actual_df, expected_df)
 
 
 @pytest.mark.parametrize(
-    "report, expected_ylims, mods", [
+    "report, expected_ylims", [
         (
             darshan.DarshanReport("examples/example-logs/ior_hdf5_example.darshan"),
             [0.0, 1.0],
-            ["POSIX", "MPI-IO", "STDIO"],
         ),
         (
             darshan.DarshanReport("examples/example-logs/sample-badost.darshan"),
             [0.0, 800.0],
-            ["POSIX", "STDIO"],
         ),
     ],
 )
-def test_plot_io_cost_ylims(report, expected_ylims, mods):
+def test_plot_io_cost_ylims(report, expected_ylims):
     # test the y limits for both axes for the IO cost stacked bar graph
 
-    for mod in mods:
-        fig = plot_io_cost(report=report, mod_key=mod)
-        for i, ax in enumerate(fig.axes):
-            # there are only 2 axes, the first being the "raw" data
-            # and the second being the normalized data (percent)
-            actual_ylims = ax.get_ylim()
-            if i == 0:
-                assert_allclose(actual_ylims, expected_ylims)
-            else:
-                # normalized data is always the same
-                assert_allclose(actual_ylims, [0.0, 100.0])
+    fig = plot_io_cost(report=report)
+    for i, ax in enumerate(fig.axes):
+        # there are only 2 axes, the first being the "raw" data
+        # and the second being the normalized data (percent)
+        actual_ylims = ax.get_ylim()
+        if i == 0:
+            assert_allclose(actual_ylims, expected_ylims)
+        else:
+            # normalized data is always the same
+            assert_allclose(actual_ylims, [0.0, 100.0])
 
 @pytest.mark.parametrize(
-    "report, expected_yticks, mods", [
+    "report, expected_yticks", [
         (
             darshan.DarshanReport("examples/example-logs/ior_hdf5_example.darshan"),
             [0.0, 0.2, 0.4, 0.6, 0.8, 1.0],
-            ["POSIX", "MPI-IO", "STDIO"],
         ),
         (
             darshan.DarshanReport("examples/example-logs/sample-badost.darshan"),
             [0, 160, 320, 480, 640, 800],
-            ["POSIX", "STDIO"],
+        ),
+        (
+            darshan.DarshanReport("tests/input/noposix.darshan"),
+            [0, 8000, 16000, 24000, 32000, 40000],
         ),
     ],
 )
-def test_plot_io_cost_y_ticks_and_labels(
-        report,
-        expected_yticks,
-        mods
-    ):
+def test_plot_io_cost_y_ticks_and_labels(report, expected_yticks):
     # check the y-axis tick marks are at the appropriate
     # locations and the labels are as expected
 
     # create the expected y-axis tick labels from the y ticks
-    expected_yticklabels = [str(i) for i in expected_yticks]
+    if np.max(expected_yticks) > 1e4:
+        expected_yticklabels = [f"{i:.0e}" for i in expected_yticks]
```

Comment:
this leads to `2.0e+4` at both 40 and 60 % ticks, which doesn't make sense. Honestly, I'd just yank out the scientific notation stuff for now so we can focus on the initial comments from Shane and Phil (i.e., use an issue for that).

Reply:
Yeah, I'll go ahead and take it out, it looks like I didn't take a close enough look at it. Probably better to address on an individual level, like you mentioned above.

Reply:
Okay, this has now been removed from the PR.